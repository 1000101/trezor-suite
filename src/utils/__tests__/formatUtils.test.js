import * as formatUtils from '../formatUtils';

describe('format utils', () => {
    it('formatAmount', () => {
        const input = [
            { amount: 0, coinInfo: { isBitcoin: true, currencyUnits: 'mbtc', shortcut: 'btc' } },
            { amount: 0.5, coinInfo: { isBitcoin: true, currencyUnits: 'mbtc', shortcut: 'btc' } },
            { amount: 1, coinInfo: { isBitcoin: false, shortcut: 'eth' } },
            { amount: 99999, coinInfo: { isBitcoin: false, currencyUnits: 'tau' } },
        ];

        input.forEach((entry) => {
            expect(formatUtils.formatAmount(entry.amount, entry.coinInfo)).toMatchSnapshot();
        });
    });

    it('formatTime', () => {
        const input = [0, 1, 2, 100, 999];

        input.forEach((entry) => {
            expect(formatUtils.formatTime(entry)).toMatchSnapshot();
        });
    });

    it('btckb2satoshib', () => {
        const input = [0, 1, 2, 100, 999];

        input.forEach((entry) => {
            expect(formatUtils.btckb2satoshib(entry)).toMatchSnapshot();
        });
    });

    it('stringToHex', () => {
        const input = ['test', '0001', 'test99999'];

        input.forEach((entry) => {
            expect(formatUtils.stringToHex(entry)).toMatchSnapshot();
        });
    });

    it('hexToString', () => {
        const input = [0, 'xxxtestSringtestStringaaaaaa', 30303031, 2746573743939393939, -9];

        input.forEach((entry) => {
            expect(formatUtils.hexToString(entry)).toMatchSnapshot();
        });
    });
});
