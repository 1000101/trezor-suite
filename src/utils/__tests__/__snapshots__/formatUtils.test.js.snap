// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format utils btckb2satoshib 1`] = `0`;

exports[`format utils btckb2satoshib 2`] = `100000`;

exports[`format utils btckb2satoshib 3`] = `200000`;

exports[`format utils btckb2satoshib 4`] = `10000000`;

exports[`format utils btckb2satoshib 5`] = `99900000`;

exports[`format utils formatAmount 1`] = `"0 btc"`;

exports[`format utils formatAmount 2`] = `"5e-9 btc"`;

exports[`format utils formatAmount 3`] = `"1e-8 eth"`;

exports[`format utils formatAmount 4`] = `"0.00099999 undefined"`;

exports[`format utils formatTime 1`] = `"No time estimate"`;

exports[`format utils formatTime 2`] = `"1 minutes"`;

exports[`format utils formatTime 3`] = `"2 minutes"`;

exports[`format utils formatTime 4`] = `"1 hour 40 minutes"`;

exports[`format utils formatTime 5`] = `"16 hours 39 minutes"`;

exports[`format utils hexToString 1`] = `""`;

exports[`format utils hexToString 2`] = `"ÂªÂªÂª"`;

exports[`format utils hexToString 3`] = `""`;

exports[`format utils hexToString 4`] = `""`;

exports[`format utils hexToString 5`] = `""`;

exports[`format utils stringToHex 1`] = `"0074006500730074"`;

exports[`format utils stringToHex 2`] = `"0030003000300031"`;

exports[`format utils stringToHex 3`] = `"007400650073007400390039003900390039"`;
